dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java

```diff
@@ -0,0 +1,168 @@
+        assertEquals("Should be a CWL workflow", Workflow.DescriptorTypeEnum.CWL, workflow2.getDescriptorType());
+        assertEquals("Should be type DOCKSTORE_YML", Workflow.ModeEnum.DOCKSTORE_YML, workflow2.getMode());
+        assertEquals("Should have one version 0.2", 1, workflow2.getWorkflowVersions().size());
+    }
+
+    /**
+     * This tests calling refresh on a workflow with a Dockstore.yml
+     */
+    @Test
+    public void testManualRefreshWorkflowWithGitHubApp() throws Exception {
+        CommonTestUtilities.cleanStatePrivate2(SUPPORT, false);
+        final ApiClient webClient = getWebClient(BasicIT.USER_2_USERNAME, testingPostgres);
+        WorkflowsApi client = new WorkflowsApi(webClient);
+
+        // Release 0.1 on GitHub - one new wdl workflow
+        List<Workflow> workflows = client.handleGitHubRelease(workflowRepo, "DockstoreTestUser2", "0.1", installationId);
+        assertEquals("Should only have one service", 1, workflows.size());
+
+        // Ensure that new workflow is created and is what is expected
+        Workflow workflow = client.getWorkflowByPath("github.com/" + workflowRepo + "/foobar", "", false);
+        assertEquals("Should be a WDL workflow", Workflow.DescriptorTypeEnum.WDL, workflow.getDescriptorType());
+        assertEquals("Should be type DOCKSTORE_YML", Workflow.ModeEnum.DOCKSTORE_YML, workflow.getMode());
+        assertEquals("Should have one version 0.1", 1, workflow.getWorkflowVersions().size());
+
+        // Refresh
+        workflow = client.refresh(workflow.getId());
+        assertNotNull(workflow);
+        assertEquals("Should have two workflow versions: 0.1, 0.2 and 0.3", 3, workflow.getWorkflowVersions().size());
+
+        Optional<WorkflowVersion> versionTwo = workflow.getWorkflowVersions().stream().filter(workflowVersion -> Objects.equals(workflowVersion.getReference(), "0.2")).findFirst();
+        Optional<WorkflowVersion> versionThree = workflow.getWorkflowVersions().stream().filter(workflowVersion -> Objects.equals(workflowVersion.getReference(), "0.3")).findFirst();
+
+        assertTrue("Version 0.2 should exist", versionTwo.isPresent());
+        assertEquals("", "/Dockstore.wdl", versionTwo.get().getWorkflowPath());
+
+        assertTrue("Version 0.3 should exist", versionThree.isPresent());
+        assertEquals("", "/Dockstore2.wdl", versionThree.get().getWorkflowPath());
+    }
+
+    /**
+     * This tests the GitHub release process does not work for users that do not exist on Dockstore
+     */
+    @Test
+    public void testGitHubReleaseNoWorkflowOnDockstoreNoUser() throws Exception {
+        CommonTestUtilities.cleanStatePrivate2(SUPPORT, false);
+        final ApiClient webClient = getWebClient(BasicIT.USER_2_USERNAME, testingPostgres);
+        WorkflowsApi client = new WorkflowsApi(webClient);
+
+        try {
+            client.handleGitHubRelease(workflowRepo, "thisisafakeuser", "0.1", installationId);
```

Comment:
missing assert fail.  test can skip apiexception and be perfectly fine.  here and other places too.